lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -89,6 +89,15 @@ def get(self, request, *args, **kwargs):
             reset_masquerade_data=True,
         )
 
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+        allow_anonymous = COURSE_ENABLE_UNENROLLED_ACCESS_FLAG.is_enabled(course_key)
+        allow_public = allow_anonymous and course.course_visibility == COURSE_VISIBILITY_PUBLIC
+        is_enrolled = enrollment and enrollment.is_active
+        is_staff = has_access(request.user, 'staff', course_key)
```

Comment:
Looking at the existing outline code, looks like we should be looking at these variables for whether we should show an outline to an unenrolled student too. But that's probably best done as a separate ticket.

Reply:
Would you mind writing up a quick ticket documenting what you mean here?

Reply:
https://openedx.atlassian.net/browse/AA-253